File: lib/src/model/get_bucket_lifecycle_result.dart
import 'bucket_lifecycle_rule.dart';
import 'oss_result.dart';

class GetBucketLifecycleResult extends OSSResult {
  List<BucketLifecycleRule>? lifecycleRules;

  void addLifecycleRule(BucketLifecycleRule? lifecycleRule) {
    lifecycleRules ??= [];
    if (lifecycleRule != null) {
      lifecycleRules!.add(lifecycleRule);
    }
  }
}

File: lib/src/model/GetBucketLifecycleResult.java
package com.alibaba.sdk.android.oss.model;

import java.util.ArrayList;

 class GetBucketLifecycleResult extends OSSResult {
     ArrayList<BucketLifecycleRule> mLifecycleRules;

     ArrayList<BucketLifecycleRule> getlifecycleRules() {
        return mLifecycleRules;
    }

     void setLifecycleRules(ArrayList<BucketLifecycleRule> lifecycleRules) {
        this.mLifecycleRules = lifecycleRules;
    }

     void addLifecycleRule(BucketLifecycleRule lifecycleRule) {
        if (mLifecycleRules == null) {
            mLifecycleRules = [];
        }

        mLifecycleRules.add(lifecycleRule);
    }
}
